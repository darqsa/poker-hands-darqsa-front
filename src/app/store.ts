import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import handsSlice from "../features/hands/slices/handsSlice";
import uiSlice from "../features/ui/slices/uiSlice";
import userSlice from "../features/users/slices/userSlice";

export const store = configureStore({
  reducer: {
    user: userSlice,
    hands: handsSlice,
    ui: uiSlice,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
