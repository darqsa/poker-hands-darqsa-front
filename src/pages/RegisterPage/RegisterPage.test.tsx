import { BrowserRouter } from "react-router-dom";
import RegisterPage from "./RegisterPage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../app/store";

describe("Given a register page component", () => {
  describe("When invoked", () => {
    test("Then it should match the snapshot", () => {
      const expectedRegisterPage = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <RegisterPage />
          </BrowserRouter>
        </Provider>
      );

      expect(expectedRegisterPage).toMatchSnapshot();
    });
  });
});
