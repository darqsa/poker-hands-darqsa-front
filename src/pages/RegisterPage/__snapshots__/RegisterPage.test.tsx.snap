// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a register page component When invoked Then it should match the snapshot 1`] = `
<form
  className="sc-gsnTZi ehTXBa form"
  onSubmit={[Function]}
>
  <h2
    className="form__heading"
  >
    Create your account
  </h2>
  <div
    className="form__group"
  >
    <label
      className="form__label"
      htmlFor="username"
    >
      Username
    </label>
    <input
      autoComplete="off"
      className="form__input"
      id="username"
      onChange={[Function]}
      placeholder="Enter your username"
      required={true}
      value=""
    />
  </div>
  <div
    className="form__group"
  >
    <label
      className="form__label"
      htmlFor="password"
    >
      Password
    </label>
    <input
      autoComplete="off"
      className="form__input "
      id="password"
      onChange={[Function]}
      placeholder="Enter your password"
      required={true}
      type="password"
      value=""
    />
  </div>
  <div
    className="form__group"
  >
    <label
      className="form__label"
      htmlFor="repeatPassword"
    >
      Repeat password
    </label>
    <input
      autoComplete="off"
      className="form__input "
      id="repeatPassword"
      onChange={[Function]}
      placeholder="Repeat your password"
      required={true}
      type="password"
      value=""
    />
  </div>
  <button
    className="sc-bczRLJ eEXWBV form__button"
    disabled={true}
    type="submit"
  >
    Register
  </button>
</form>
`;
